models/Site.php

```diff
@@ -220,11 +220,16 @@ public function setId(int $id): static
     /**
      * @return $this
      */
-    public function setDomains(mixed $domains): static
+    public function setDomains(array|string $domains): static
     {
         if (is_string($domains)) {
             $domains = \Pimcore\Tool\Serialize::unserialize($domains);
         }
+        array_map(static function ($domain) {
+            if (!filter_var($domain, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME)) {
+                throw new \InvalidArgumentException(sprintf('Invalid domain name "%s"', $domain));
```

Comment:
We should import the exception :)

Reply:
@mattamon Done in https://github.com/pimcore/pimcore/pull/17221/commits/c986aea020478bc928b5ccdfe390495b8e214c1c